docs/dev/devel-env.rst

```diff
@@ -286,12 +294,152 @@ Also, run the **integration tests** for ListenBrainz.
 When the tests complete, you will see if your changes are valid or not. These tests
 are a helpful way to validate new changes without a lot of work.
 
-FAQ's
+Set up Listenbrainz Spark development environment
+=================================================
+
+The Listenbrainz Spark module is used to generate recommendations and stats using Apache Spark. The recommendations are generated based on collaborative filtering technique.
+
+To contribute to the ListenBrainz Spark project, you need a development environment.
+With your development environment, you can test your changes before submitting a
+patch back to the project. This guide helps you set up a development environment
+and run ListenBrainz Spark locally on your workstation. By the end of this guide, you
+will have…
+
+* Install system dependencies
+* Run ListenBrainz Spark locally on your workstation
+
+
+Install dependencies
+--------------------
+
+The ``listenbrainz_spark-server`` communicates with the ``listenbrainz-server``. 
+So you will need to set up the ``listenbrainz-server`` using the above-given steps.
+
+The ``listenbrainz_spark-server`` is shipped in Docker containers just like the 
+``listenbrainz-server``. This helps create your development environment and later deploy the application. Therefore, to work on the project, you need to install Docker
+and use containers for building the project. Containers save you from installing all of this on your own workstation.
+
+For different Docker installation instructions for your distribution refer the steps given above.
+
+Update config.py
+^^^^^^^^^^^^^^^^
```

Comment:
is this intended? shouldn't it be dashed?

Reply:
I will have to check. I think its for subheadings too like dashed

Reply:
This gives sub-sub-heading 
=== H1
----- H2
^^^^ H3

Reply:
cool :)